lib/rubocop/cop/base.rb

```diff
@@ -430,7 +430,7 @@ def file_name_matches_any?(file, parameter, default_result)
         return default_result unless patterns
 
         patterns = FilePatterns.from(patterns)
-        patterns.match?(config.path_relative_to_config(file)) || patterns.match?(file)
+        patterns.match?(config.path_relative_to_config(file)) || patterns.match?(file) || patterns.match?(Pathname.new(file).relative_path_from(Pathname.pwd).to_s)
```

Comment:
I added this because if we have an include like this:
```yml
Style/SomeCop:
  Include:
    - packs/**/*
```

And a file:
```
Pathname.pwd.join('packs/my_pack/blah.rb')
# /Users/me/path/to/directory/packs/my_pack/blah.rb
```

Then that `Include` wouldn't match that path otherwise.

To put it another way – the first match check checks if the include matches the file when the file is relative to the directory where the config file lives. The second does a straight match, which would work here if the path were absolute, and lastly we need to check if the pattern matches the file relative to the working directory, which mirrors the change above to change `Include` paths to be relative to the `pwd`.

Note that if the `file` doesn't have the same base as `Pathname.pwd`, we get this error:
```
     ArgumentError:
       different prefix: "" and "/Users/alex.evanczuk/workspace/rubocop"
```

However, I don't think it's supported in rubocop to run rubocop against a file that is *not* in the `Pathname.pwd`, so an error seems to make sense here.

When I try to `bundle exec rubocop /path/to/file/in/other/repo`, it leads to strange and unpredictable errors. Therefore, I changed the tests to match to reflect the fact that they should be considered in the same pwd as the one rubocop runs in.

Reply:
I went ahead and rescued this error, for two reasons:
1) In the specs for `windows`, `create_file` was creating a file in a directory that was different than `Dir.pwd`. We should probably fix this, but it made me realize that...
2) In the case where folks have valid use cases for running rubocop on a file in a directory different than the `Dir.pwd`, rather than breaking their workflows, we can simply `rescue` which would fall back to the existing behavior.

Reply:
I don't really understand why the include path in the `.rubocop.yml` you give at the beginning of your comment would have to match paths relative to the current working directory. These path patterns should only be interpreted as relative to the configuration file's directory IMO.

Maybe you have a point that I was unable to grasp, but I tried reverting `base.rb` back to its previous state, i.e. I removed the last part of the condition, and all specs still passed. This means that we don't have a spec that shows why this logic needs to be changed.

Reply:
Thanks @jonas054 , I see what you're saying, and you're exactly right here. I reverted it as well and ran it fully integrated on our application's codebase and found that I didn't need it. I've been on vacation so I forgot why I thought I needed it, but it looks like as you said this is not necessary.

Let me know if you have any other feedback about this PR!